File: 2025/Day1/advent_of_code_001_001.py
with open("input/2025-1") as file:
    rotations = [line.strip() for line in file]

dial_position = 50
total_zeros = 0

for rotation in rotations:
    direction, distance = rotation[0], int(rotation[1:])

    if direction == "L":
        dial_position -= distance       
    else: 
        dial_position += distance

    dial_position %= 100

    if dial_position == 0:
        total_zeros += 1

print(f"total zeros: {total_zeros}")

File: 2025/Day1/cpp/advent_of_code_001_001.cpp
#include <fstream>
#include <iostream>
#include <string>


int main() {
    std::ifstream input_file("../input/2025-1");

    if (!input_file.is_open()) {
        std::cerr << "Failed to open" << "\n";
        return 1;
    }

    std::string rotation{};
    int dial_position = 50;
    int total_zeros = 0;

    while(getline(input_file, rotation)) {
        char direction = rotation[0];
        int distance = std::stoi(rotation.substr(1));

        if (direction == 'L') {
            dial_position -= distance;
        }
        else {
            dial_position += distance;
        }

        dial_position = (dial_position % 100 + 100) % 100;

        if (dial_position == 0) {
            total_zeros += 1;
        }
    }
    std::cout << "Total zeros: " << total_zeros << "\n";

    return 0;
}